### ISSUE TITLE
Allow using `getApplicationSupportDirectory` instead of `getApplicationDocumentsDirectory` in drift_flutter

### ISSUE THREAD

**Issue by clragon** [simolus3/moor#3139]
**Is your feature request related to a problem? Please describe.**
My app generally uses the Application Support Directory for the database.
This is because the database isn't considered to be a file that is "exposed" to the user.
On Windows, this makes the difference between the AppData directory (✅) and the user documents directory (❌).

**Describe the solution you'd like**
I am not sure how customizable this should be. personally I am fine with just being able to flip a boolean to use the Support Directory instead of the Documents Directory but maybe this could be a chance to allow customizing the path further.

**Comment by simolus3** [comment]
Good call! bc6e1a75da0916c3baf8a244476d3e67a149e58a adds an optional callback which can asynchronously create the database path to use.

**Comment by JulienElkaim** [comment]
Thanks @clragon  !

Just caught this issue as well today.
Waiting for version 2.22 to be in prod (Now in dev only right?), what do you recommend as workaround simolus?

I do smth like: 

```dart  
class AppDatabase extends _$AppDatabase {
 
  AppDatabase() : super(openConnection());

  @override
  int get schemaVersion => 1;

  static LazyDatabase openConnection() {
    final databaseName ="my_database";
    return LazyDatabase(() async {
      final appSupportDir = await getApplicationSupportDirectory();
      final dbFile = File(join(appSupportDir.path, 'databases', '$databaseName.sqlite'));

      return NativeDatabase(dbFile);
    });
  }
}
```

**Comment by simolus3** [comment]
The changes have been released with `drift_flutter` version 0.2.0 and should be available since that release.
